File: engine/model/src/main/scala/tools/packet/transfer/name/CheckNameChange.scala
package tools.packet.transfer.name

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class CheckNameChange(private var _availableName: String, private var _canUseName: Boolean) extends PacketInput {
  def availableName: String = _availableName

  def canUseName: Boolean = _canUseName

  override def opcode(): SendOpcode = SendOpcode.CASHSHOP_CHECK_NAME_CHANGE
}

File: engine/engine-model/src/main/java/tools/packet/transfer/name/CheckNameChange.java
package tools.packet.transfer.name;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record CheckNameChange(String availableName, Boolean canUseName) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.CASH_SHOP_CHECK_NAME_CHANGE;
   }
}